src/services/NotifierClient.cpp

```diff
@@ -0,0 +1,29 @@
+#include "./services/NotifierClient.h"
+
+NotifierClient::NotifierClient(const std::string publisher) : publisher{publisher}
+{
+}
+
+std::pair<std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds>, Notification> NotifierClient::createNotification(
+    const std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> timeToFire,
+    const std::string message, const std::array<bool, 7> daysOfWeek)
```

Comment:
Warning: With `const std::string message` you are making a copy of the string itself - try to use `const std::string& message` instead! The `const` modifier won't help you here (I think).

Reply:
You are right, for the arguments passed by values `const` modifier is useless. We will remove these unnecessary `const` from now on (we have a few of them in the rest of our code).

However, I would argue that in modern C++ we should always pass by value instead of `const &` (I would only use `const &` for very large objects). Source: https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference

Reply:
Oh - according to https://stackoverflow.com/a/51706522/11664229, this is a valid point. I would still argue as most classes of the OS / apps do not even implement the copy-constructor correctly, just hoping that the default move-constructor works in a more intelligent way is a bad bet. Therefore preventing two copies in the worst case, I would still take the one-copy in every case version. Still, the `std`-libs should work correctly, so feel free to treat this accordingly (I'll try to better myself).